src/main/java/sirius/db/mixing/EntityDescriptor.java

```diff
@@ -1045,6 +1047,40 @@ public Property getProperty(String property) {
         return prop;
     }
 
+    /**
+     * Get the value of a mapping for an entity.
+     * <p>
+     * If the mapping contains joins, they are only resolved if (a) they are already fetched from the database, or (b)
+     * the parameter value of {@code resolveJoins} evaluates to {@code true}. In all other cases, an Exception is thrown.
+     * <p>
+     * Usually, this can be achieved using <p>
+     * {@link #getProperty(Mapping)}{@code .}{@link Property#getValue(Object) getValue(Object)}<p>
+     * However, no property can be created if the mapping is a JOIN-mapping, rendering this method useless in that case.
+     *
+     * @param mapping      the mapping; may contain joins
+     * @param entity       the entity
+     * @param resolveJoins whether to load joined columns from the database instead of throwing
+     * @return the value of the mapping
+     */
+    public Object getPropertyValue(Mapping mapping, Object entity, boolean resolveJoins) {
+        if (mapping.getParent() != null) {
+            Object parent = getPropertyValue(mapping.getParent(), entity, resolveJoins);
+            return mixing.getDescriptor(parent.getClass())
+                         .getPropertyValue(Mapping.named(mapping.getName()), parent, resolveJoins);
+        }
+        Property property = getProperty(mapping.getName());
+        if (property instanceof BaseEntityRefProperty<?, ?, ?> ref) {
+            BaseEntityRef<?, ?> entityRef = ref.getEntityRef(entity);
+            return resolveJoins ? entityRef.fetchValue() : entityRef.getValueIfPresent().orElseThrow(() -> {
+                return new IllegalArgumentException(Strings.apply(
+                        "The BaseEntityRef `%s` is not loaded, but is requested by the mapping `%s`.",
+                        entityRef.getUniqueObjectName(),
+                        mapping));
+            });
+        }
+        return property.getValue(entity);
+    }
+
     /**
```

Comment:
Keep this as private method in SmartQuery, kill resolveJoins as it is never used - so probably we can just walk through mapping.getParent and repeatedly call getProperty().getValue() on the proper descriptors - so this all should be quite a lot simpler... On the actual call site, we know for sure, that the value is fetched, so no special checks are needed either...